lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,105 @@
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		
+		static get observedAttributes() { return ['data-id']};
+		
+		constructor () {
+			super();
+			
+			// Hey Eric! I have no clue if ive done this corret at all, but I am still extremly confused about this whole setup
+			// We have two constructors for different purposes. Im not sure if i implemented the view correctly.
+			
+			var id = this.getAttribute("data-id");  
+			var descriptionArray  = ["Take ten normal cats and combine them into a Recruiter! "
+				+"Can produce 1 CATS for every 15 seconds.", "Take thirty normal cats and combine them into a Trainer! "
+				+" Produces 5 CATS for every 15 seconds.", "Take 75 cats and hire them to run a boot camp. What could go wrong?"
+				+"  Produces 15 CATS for every 15 seconds"]; 		
+			var nameArray =[ "Recrutier","Trainer" ,"Camp"];
+			var totalGenArray = ["4","20","60"];
+			var costArray = ["10","30","75"];
+			
+			
+			
+			var name = nameArray[id];
+			var description = descriptionArray[id];
+			var totalGen = totalGenArray[id];
+			var cost =costArray[id];
+			
+			
+			console.log()
+			
+			var shadowRoot = this.attachShadow({mode: 'open'});
+			var wrapper = document.createElement('form');
+			wrapper.innerHTML = ` 
+			<link rel="stylesheet" href="app.css">
+			
+			<form>
+			
+					<div class= "generator">
+					<p class="amount">0</p>
+					<h5>CATS ${name}</h5>
+					<p>${description}</p>
+					<input type="button" value="${cost} Resource" class="recourse"></button>
```

Comment:
Cost should be dynamic/computed value from the `getCost` function implemented in the Generator model.